salt/modules/dockerng.py

```diff
@@ -5529,3 +5541,244 @@ def script_retcode(name,
                    ignore_retcode=ignore_retcode,
                    use_vt=use_vt,
                    keep_env=keep_env)['retcode']
+
+
+def _mk_fileclient():
+    '''
+    Create a file client and add it to the context.
+    '''
+    if 'cp.fileclient' not in __context__:
+        __context__['cp.fileclient'] = \
+                salt.fileclient.get_file_client(__opts__)
+
+
+def _generate_tmp_path():
+    return os.path.join(
+        '/tmp',
+        'salt.dockerng.{0}'.format(uuid.uuid4().hex[:6]))
+
+
+def _prepare_trans_tar(mods=None, saltenv='base', pillar=None):
+    '''
+    Prepares a self contained tarball that has the state
+    to be applied in the container
+    '''
+    chunks = _compile_state(mods, saltenv)
+    # reuse it from salt.ssh, however this function should
+    # be somewhere else
+    refs = salt.client.ssh.state.lowstate_file_refs(chunks)
+    _mk_fileclient()
+    trans_tar = salt.client.ssh.state.prep_trans_tar(
+        __context__['cp.fileclient'],
+        chunks, refs, pillar=pillar)
```

Comment:
We still need an id passed in here. Otherwise we're getting this:

```
AttributeError: 'NoneType' object has no attribute 'startswith'
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/home/mp/devel/salt/salt/scripts.py", line 352, in salt_call
    client.run()
  File "/home/mp/devel/salt/salt/cli/call.py", line 58, in run
    caller.run()
  File "/home/mp/devel/salt/salt/cli/caller.py", line 134, in run
    ret = self.call()
  File "/home/mp/devel/salt/salt/cli/caller.py", line 197, in call
    ret['return'] = func(*args, **kwargs)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5780, in sls_build
    __salt__['dockerng.sls'](id_, mods, saltenv, **kwargs)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5711, in sls
    trans_tar = _prepare_trans_tar(mods=mods, saltenv=saltenv, pillar=pillar)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5573, in _prepare_trans_tar
    chunks, refs, pillar=pillar)
  File "/home/mp/devel/salt/salt/client/ssh/state.py", line 148, in prep_trans_tar
    cachedir = os.path.join('salt-ssh', id_)
  File "/usr/lib/python2.7/posixpath.py", line 68, in join
    if b.startswith('/'):
AttributeError: 'NoneType' object has no attribute 'startswith'
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/home/mp/devel/salt/salt/scripts.py", line 352, in salt_call
    client.run()
  File "/home/mp/devel/salt/salt/cli/call.py", line 58, in run
    caller.run()
  File "/home/mp/devel/salt/salt/cli/caller.py", line 134, in run
    ret = self.call()
  File "/home/mp/devel/salt/salt/cli/caller.py", line 197, in call
    ret['return'] = func(*args, **kwargs)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5780, in sls_build
    __salt__['dockerng.sls'](id_, mods, saltenv, **kwargs)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5711, in sls
    trans_tar = _prepare_trans_tar(mods=mods, saltenv=saltenv, pillar=pillar)
  File "/home/mp/devel/salt/salt/modules/dockerng.py", line 5573, in _prepare_trans_tar
    chunks, refs, pillar=pillar)
  File "/home/mp/devel/salt/salt/client/ssh/state.py", line 148, in prep_trans_tar
    cachedir = os.path.join('salt-ssh', id_)
  File "/usr/lib/python2.7/posixpath.py", line 68, in join
    if b.startswith('/'):
AttributeError: 'NoneType' object has no attribute 'startswith'
```